### ISSUE TITLE
Package haskell-language-server does not install 

### ISSUE THREAD

**Issue by siddhantdev** [williamboman/mason.nvim#1463]
### I've searched open issues for similar requests

- [X] Yes

### I've recently downloaded the latest plugin version of mason.nvim

- [X] Yes

### Problem description

Installing haskell-language-server via Mason fails.

### Expected behavior

The package should install. However trying to install gives an error.

### Affected packages

haskell-language-server

### Mason output

```Text
+ ghcup install hls 2.0.0.0 -i /home/siddhant/.local/share/nvim/mason/packages/haskell-language-server
        Invalid argument `2.0.0.0'
        
        Usage: ghcup [-v|--verbose] [-c|--cache] [-n|--no-verify] COMMAND
        
        Available options:
          -v,--verbose             Enable verbosity
          -c,--cache               Cache downloads in ~/.ghcup/cache
          -n,--no-verify           Skip tarball checksum verification
          --keep <always|errors|never>
                                   Keep build directories? (default: never)
          --downloader <curl|wget> Downloader to use (default: curl)
          -h,--help                Show this help text
          --version                Show version
          --numeric-version        Show the numeric version (for use in scripts)
        
        Main commands:
          tui                      Start the interactive GHCup UI
          install                  Install or update GHC/cabal
          set                      Set currently active GHC/cabal version
          rm                       Remove a GHC/cabal version
          list                     Show available GHCs and other tools
          upgrade                  Upgrade ghcup
          compile                  Compile a tool from source
        
        Other commands:
          debug-info               Show debug info
          tool-requirements        Show the requirements for ghc/cabal
          changelog                Find/show changelog
        
        Discussion:
          ghcup installs the Glasgow Haskell Compiler from the official
          release channels, enabling you to easily switch between different
          versions. It maintains a self-contained ~/.ghcup directory.
        
        ENV variables:
          * TMPDIR: where ghcup does the work (unpacking, building, ...)
          * GHCUP_INSTALL_BASE_PREFIX: the base of ghcup (default: $HOME)
        
        Report bugs at <https://gitlab.haskell.org/haskell/ghcup-hs/issues>
        spawn: bash failed with exit code 1 and signal 0.
```

### Installation log

```Text
[DEBUG Mon 21 Aug 2023 06:41:48 PM IST] ...ocal/share/nvim/lazy/mason.nvim/lua/mason-core/fetch.lua:37: Fetching URL "https://api.mason-registry.dev/api/github/mason-org/mason-registry/releases/latest"
[DEBUG Mon 21 Aug 2023 06:41:48 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="curl", spawn_opts={
  args = { "-H", "Accept: application/vnd.mason-registry.v1+json; q=1.0, application/json; q=0.8", "-H", "User-Agent: mason.nvim v1.6.2 (+https://github.com/williamboman/mason.nvim)", "-fsSL", "-X", "GET", "--connect-timeout", 30, "https://api.mason-registry.dev/api/github/mason-org/mason-registry/releases/latest" }
}
[DEBUG Mon 21 Aug 2023 06:41:48 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 8881
[DEBUG Mon 21 Aug 2023 06:41:53 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:148: Job pid=8881 exited with exit_code=6, signal=0
[DEBUG Mon 21 Aug 2023 06:41:53 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="wget", spawn_opts={
  args = { "--header=Accept: application/vnd.mason-registry.v1+json; q=1.0, application/json; q=0.8", "--header=User-Agent: mason.nvim v1.6.2 (+https://github.com/williamboman/mason.nvim)", "-nv", "-o", "/dev/null", "-O", "-", "--timeout=30", "--method=GET", "https://api.mason-registry.dev/api/github/mason-org/mason-registry/releases/latest" }
}
[DEBUG Mon 21 Aug 2023 06:41:53 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 8883
[INFO  Mon 21 Aug 2023 06:41:54 PM IST] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:184: Executing installer for Package(name=haskell-language-server) {}
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:58: Attempting to lock package Package(name=haskell-language-server)
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:70: Wrote lockfile Package(name=haskell-language-server)
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:71: fs: mkdirp /home/siddhant/.local/share/nvim/mason/staging/haskell-language-server
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] ...zy/mason.nvim/lua/mason-core/installer/registry/init.lua:148: Compiling installer. haskell-language-server {}
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] ...zy/mason.nvim/lua/mason-core/installer/registry/init.lua:109: Parsing spec haskell-language-server {}
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] ...y/mason.nvim/lua/mason-core/installer/managers/build.lua:22: build: run {
  bin = {
    server_8_10_7 = "bin/haskell-language-server-8.10.7",
    server_9_0_2 = "bin/haskell-language-server-9.0.2",
    server_9_2_7 = "bin/haskell-language-server-9.2.7",
    server_9_4_5 = "bin/haskell-language-server-9.4.5",
    server_9_6_1 = "bin/haskell-language-server-9.6.1",
    wrapper = "bin/haskell-language-server-wrapper"
  },
  env = {
    VERSION = "2.0.0.0"
  },
  run = 'ghcup install hls "$VERSION" -i "$PWD"',
  staged = false,
  target = "unix"
}
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] ...vim/lazy/mason.nvim/lua/mason-core/installer/context.lua:229: Promoting cwd "/home/siddhant/.local/share/nvim/mason/staging/haskell-language-server" to "/home/siddhant/.local/share/nvim/mason/packages/haskell-language-server"
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:65: fs: mkdir /home/siddhant/.local/share/nvim/mason/packages/haskell-language-server
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:84: fs: rename /home/siddhant/.local/share/nvim/mason/staging/haskell-language-server /home/siddhant/.local/share/nvim/mason/packages/haskell-language-server
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="bash", spawn_opts={
  args = {},
  cwd = "/home/siddhant/.local/share/nvim/mason/packages/haskell-language-server",
  env = { "LS_COLORS=<redacted>", "GTK_MODULES=<redacted>", "ALACRITTY_LOG=<redacted>", "OWD=<redacted>", "LOGNAME=<redacted>", "DBUS_SESSION_BUS_ADDRESS=<redacted>", "DISPLAY=<redacted>", "XDG_RUNTIME_DIR=<redacted>", "DEFAULTS_PATH=<redacted>", "GPG_AGENT_INFO=<redacted>", "XDG_CURRENT_DESKTOP=<redacted>", "SHLVL=<redacted>", "VIM=<redacted>", "DESKTOP_SESSION=<redacted>", "SSH_AUTH_SOCK=<redacted>", "GTK_IM_MODULE=<redacted>", "ALACRITTY_SOCKET=<redacted>", "APPIMAGE=<redacted>", "USER=<redacted>", "XDG_MENU_PREFIX=<redacted>", "NVIM_LOG_FILE=<redacted>", "GTK3_MODULES=<redacted>", "GDMSESSION=<redacted>", "P9K_TTY=<redacted>", "SHELL=<redacted>", "APPDIR=<redacted>", "SESSION_MANAGER=<redacted>", "GNOME_DESKTOP_SESSION_ID=<redacted>", "_P9K_TTY=<redacted>", "COLORTERM=<redacted>", "OLDPWD=<redacted>", "WINDOWID=<redacted>", "MYVIMRC=<redacted>", "USERNAME=<redacted>", "XDG_CONFIG_DIRS=<redacted>", "WINDOWPATH=<redacted>", "ALACRITTY_WINDOW_ID=<redacted>", "XDG_SESSION_CLASS=<redacted>", "XMODIFIERS=<redacted>", "VIMRUNTIME=<redacted>", "GIO_LAUNCHED_DESKTOP_FILE_PID=<redacted>", "XAUTHORITY=<redacted>", "SYSTEMD_EXEC_PID=<redacted>", "HOME=<redacted>", "QT_ACCESSIBILITY=<redacted>", "XDG_DATA_DIRS=<redacted>", "_P9K_SSH_TTY=<redacted>", "SSH_AGENT_LAUNCHER=<redacted>", "LANG=<redacted>", "JOURNAL_STREAM=<redacted>", "P9K_SSH=<redacted>", "MANDATORY_PATH=<redacted>", "XDG_SESSION_TYPE=<redacted>", "INVOCATION_ID=<redacted>", "TERM=<redacted>", "XDG_SESSION_DESKTOP=<redacted>", "PATH=/home/siddhant/.local/share/nvim/mason/bin:/home/siddhant/.local/share/nvim/mason/bin:/home/siddhant/bin/flutter/bin:/home/siddhant/bin:/home/siddhant/.local/bin:/home/siddhant/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/siddhant/.local/bin", "MANAGERPID=<redacted>", "MASON=<redacted>", "GNOME_SHELL_SESSION_MODE=<redacted>", "PWD=<redacted>", "QT_IM_MODULE=<redacted>", "EDITOR=<redacted>", "VERSION=<redacted>" }
}
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 8884
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:148: Job pid=8884 exited with exit_code=1, signal=0
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:59: fs: unlink /home/siddhant/.local/share/nvim/mason/staging/haskell-language-server.lock
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] .../.local/share/nvim/lazy/mason.nvim/lua/mason-core/fs.lua:47: fs: rmrf /home/siddhant/.local/share/nvim/mason/packages/haskell-language-server
[DEBUG Mon 21 Aug 2023 06:41:54 PM IST] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:39: Building receipt for Package(name=haskell-language-server)
[ERROR Mon 21 Aug 2023 06:41:54 PM IST] ...e/nvim/lazy/mason.nvim/lua/mason-core/installer/init.lua:249: Installation failed for Package(name=haskell-language-server) error=spawn: bash failed with exit code 1 and signal 0. 
[DEBUG Mon 21 Aug 2023 06:41:58 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:148: Job pid=8883 exited with exit_code=4, signal=0
[ERROR Mon 21 Aug 2023 06:41:58 PM IST] ...e/nvim/lazy/mason.nvim/lua/mason-core/providers/init.lua:75: Provider "github" "get_latest_release" failed: spawn: wget failed with exit code 4 and signal 0. 
[DEBUG Mon 21 Aug 2023 06:41:58 PM IST] ...ocal/share/nvim/lazy/mason.nvim/lua/mason-core/spawn.lua:80: "gh" is not executable
[DEBUG Mon 21 Aug 2023 06:41:58 PM IST] ...ocal/share/nvim/lazy/mason.nvim/lua/mason-core/fetch.lua:37: Fetching URL "https://api.github.com/repos/mason-org/mason-registry/releases/latest"
[DEBUG Mon 21 Aug 2023 06:41:58 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="curl", spawn_opts={
  args = { "-H", "Accept: application/vnd.github.v3+json; q=1.0, application/json; q=0.8", "-H", "User-Agent: mason.nvim v1.6.2 (+https://github.com/williamboman/mason.nvim)", "-fsSL", "-X", "GET", "--connect-timeout", 30, "https://api.github.com/repos/mason-org/mason-registry/releases/latest" }
}
[DEBUG Mon 21 Aug 2023 06:41:58 PM IST] ...al/share/nvim/lazy/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 8890
```

### Neovim version (>= 0.7)

NVIM v0.9.1
Build type: Release
LuaJIT 2.1.0-beta3

### Operating system/version

Linux pop-os 6.4.6-76060406-generic #202307241739~1690928105~22.04~d567a38 SMP PREEMPT_DYNAMIC Tue A x86_64 x86_64 x86_64 GNU/Linux

### Healthcheck

```Text
mason: require("mason.health").check()

mason.nvim ~
- OK mason.nvim version v1.6.2
- OK PATH: prepend
- OK Providers: 
  mason.providers.registry-api
  mason.providers.client
- OK neovim version >= 0.7.0

mason.nvim [Registries] ~
- OK Registry `github.com/mason-org/mason-registry version: 2023-08-21-winged-magic` is installed.
- OK Registry `github.com/mason-org/mason-registry version: 2023-08-21-winged-magic` is installed.

mason.nvim [Core utils] ~
- OK unzip: `UnZip 6.00 of 20 April 2009, by Debian. Original by Info-ZIP.`
- OK wget: `GNU Wget 1.21.2 built on linux-gnu.`
- OK curl: `curl 7.81.0 (x86_64-pc-linux-gnu) libcurl/7.81.0 OpenSSL/3.0.2 zlib/1.2.11 brotli/1.0.9 zstd/1.4.8 libidn2/2.3.2 libpsl/0.21.0 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.43.0 librtmp/2.3 OpenLDAP/2.5.16`
- OK gzip: `gzip 1.10`
- OK tar: `tar (GNU tar) 1.34`
- OK bash: `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
- OK sh: `Ok`

mason.nvim [Languages] ~
- WARNING Go: not available
  - ADVICE:
    - spawn: go failed with exit code - and signal -. go is not executable
- WARNING cargo: not available
  - ADVICE:
    - spawn: cargo failed with exit code - and signal -. cargo is not executable
- WARNING luarocks: not available
  - ADVICE:
    - spawn: luarocks failed with exit code - and signal -. luarocks is not executable
- WARNING Composer: not available
  - ADVICE:
    - spawn: composer failed with exit code - and signal -. composer is not executable
- WARNING PHP: not available
  - ADVICE:
    - spawn: php failed with exit code - and signal -. php is not executable
- WARNING javac: not available
  - ADVICE:
    - spawn: javac failed with exit code - and signal -. javac is not executable
- OK node: `v18.17.1`
- WARNING julia: not available
  - ADVICE:
    - spawn: julia failed with exit code - and signal -. julia is not executable
- OK Ruby: `ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux-gnu]`
- OK python: `Python 3.10.12`
- OK java: `openjdk version "18.0.2-ea" 2022-07-19`
- OK RubyGem: `3.3.5`
- OK npm: `9.6.7`
- OK pip: `pip 23.2.1 from /home/siddhant/.local/lib/python3.10/site-packages/pip (python 3.10)`
- OK python venv: `Ok`

mason.nvim [GitHub] ~
- OK GitHub API rate limit. Used: 0. Remaining: 60. Limit: 60. Reset: Mon 21 Aug 2023 07:44:20 PM IST.
  Install and authenticate via gh-cli to increase rate limit.
```

**Comment by williamboman** [comment]
Hello! What version of `ghcup` do you have installed?

**Comment by siddhantdev** [comment]
Running `ghcup --version` gives me the following output:

```
The GHCup Haskell installer, version 0.1.19.4
```

**Comment by siddhantdev** [comment]
Ok, so the version comment got me digging a bit further on that regard. Turns out there was some discrepancy between recommended version of ghcup and latest version of ghcup being installed and which one was being used due to the way my path was setup. After ensuring that the latest version of both ghcup and ghc were being used, I was able to get hls to install through Mason. Thanks a lot for your help.
